### PR TITLE
TLS support added

### PR DESCRIPTION
**Pull request by justinma246**
TLS support added by converting tls to ssl internally
Ticket https://github.com/kaazing/gateway/issues/602

### CODE REVIEW

File: server/src/test/java/org/kaazing/gateway/server/context/resolve/TLStoSSLTest.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+import java.text.MessageFormat;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.kaazing.gateway.server.test.Gateway;
```
Comment (msalavastru) [inline server/src/test/java/org/kaazing/gateway/server/context/resolve/TLStoSSLTest.java]: Please remove unused imports.

File: server/src/test/java/org/kaazing/gateway/server/context/resolve/TLStoSSLTest.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        int datum;
+        while ((datum = is.read()) != -1) {
+            fos.write(datum);
+        }
+        fos.flush();
+        fos.close();
+        return file;
+    }
+    
+    private File createTempFileFromResource(String resourceName, String... values) throws IOException {
```
Comment (msalavastru) [inline server/src/test/java/org/kaazing/gateway/server/context/resolve/TLStoSSLTest.java]: Please remove this method if it is never used.

File: server/src/test/java/org/kaazing/gateway/server/context/resolve/TLStoSSLTest.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+    @Test
+    public void parseAndResolveTLSinAccept() throws Exception {
+        configFile = createTempFileFromResource(
+                "org/kaazing/gateway/server/config/parse/data/gateway-config-tls-in-accept.xml");
+        GatewayConfigDocument doc = parser.parse(configFile);
+        Assert.assertNotNull(doc);
+        resolver.resolve(doc);
+    }
+
```
Comment (msalavastru) [inline server/src/test/java/org/kaazing/gateway/server/context/resolve/TLStoSSLTest.java]: Remove this tab. You should use the Kaazing code formatter to format the contents of this file.

File: server/src/test/java/org/kaazing/gateway/server/context/resolve/TLStoSSLTest.java [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+                "org/kaazing/gateway/server/config/parse/data/gateway-config-tls-in-accept.xml");
+        GatewayConfigDocument doc = parser.parse(configFile);
+        Assert.assertNotNull(doc);
+        resolver.resolve(doc);
+    }
+    
+    @Test
+    public void parseAndResolveTLSAcceptOptions() throws Exception {
+        configFile = createTempFileFromResource(
+                "org/kaazing/gateway/server/config/parse/data/gateway-config-tls-accept-options.xml");
```
Comment (msalavastru) [inline server/src/test/java/org/kaazing/gateway/server/context/resolve/TLStoSSLTest.java]: Here you can use String constants such as:

```
    private static final String BASE_PATH = "org/kaazing/gateway/server/config/parse/data/";
    private static final String ACCEPT_OPTIONS_XML = "gateway-config-tls-accept-options.xml";
    ...
    configFile = createTempFileFromResource(String.format("%s%s", BASE_PATH, ACCEPT_OPTIONS_XML));

```

File: server/src/test/java/org/kaazing/gateway/server/context/resolve/TLStoSSLTest.java [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+public class TLStoSSLTest {
+
+    private static GatewayConfigParser parser;
+    private static GatewayContextResolver resolver;
+
+    private File configFile;
+    private File keyStoreFile;
+    private File keyStorePasswordFile;
+    private File trustStoreFile;
```
Comment (msalavastru) [inline server/src/test/java/org/kaazing/gateway/server/context/resolve/TLStoSSLTest.java]: It looks like these are not used anywhere (keyStoreFile, keyStorePasswordFile, trustStoreFile).

File: server/src/test/java/org/kaazing/gateway/server/context/resolve/TLStoSSLTest.java [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+        return file;
+    }
+
+    @Test
+    public void parseAndResolveTLSinAccept() throws Exception {
+        configFile = createTempFileFromResource(
+                "org/kaazing/gateway/server/config/parse/data/gateway-config-tls-in-accept.xml");
+        GatewayConfigDocument doc = parser.parse(configFile);
+        Assert.assertNotNull(doc);
+        resolver.resolve(doc);
```
Comment (msalavastru) [inline server/src/test/java/org/kaazing/gateway/server/context/resolve/TLStoSSLTest.java]: What do you check here?

**Comment by msalavastru** [comment]
@justinma246 I have added a few comments above with what I found so far. I am thinking you can also add some integration tests with k3po that use tls in different scenarios.

File: server/src/test/java/org/kaazing/gateway/server/context/resolve/TLStoSSLTest.java (hunk 6 above)
Comment (justinma246) [inline server/src/test/java/org/kaazing/gateway/server/context/resolve/TLStoSSLTest.java]: I am checking to see if 'tls' in a gateway-config can be parsed and resolved with no error

**Comment by jitsni** [comment]
It is part of the PR https://github.com/kaazing/gateway/issues/788

**Pull request merged**